### PR TITLE
Containers: Test for google kubernetes engine

### PR DESCRIPTION
**Pull request by ilausuch**
- Related ticket: https://progress.opensuse.org/issues/97721
- Verification run: http://marvin5.arch.suse.cz/tests/212

### CODE REVIEW

File: lib/publiccloud/basetest.pm [hunk 1]
```diff
@@ -72,15 +73,10 @@ sub provider_factory {
...
         $args{service} //= 'GCE';
         if ($args{service} eq 'GCR') {
-            $provider = publiccloud::gcr->new(
-                account => get_var('PUBLIC_CLOUD_GOOGLE_ACCOUNT'),
-                service_acount_name => get_var('PUBLIC_CLOUD_GOOGLE_SERVICE_ACCOUNT'),
-                project_id => get_var('PUBLIC_CLOUD_GOOGLE_PROJECT_ID'),
-                client_id => get_var('PUBLIC_CLOUD_GOOGLE_CLIENT_ID'),
-                region => get_var('PUBLIC_CLOUD_REGION', 'europe-west1-b'),
-                storage_name => get_var('PUBLIC_CLOUD_GOOGLE_STORAGE', 'openqa-storage'),
-                username => get_var('PUBLIC_CLOUD_USER', 'susetest')
```
Comment (pdostal) [inline lib/publiccloud/basetest.pm:82]: Why did you remove those parameters?

File: lib/publiccloud/gcp_client.pm [hunk 2]
```diff
@@ -121,7 +122,7 @@ Get the full registry prefix URL for any containers image registry of ECR based
 
 sub get_container_registry_prefix {
     my ($self) = @_;
-    return sprintf($self->gcr_zone . '/suse-sle-qa', $self->project_id);
+    return $self->gcr_zone . '/' . $self->project_id;
```
Comment (pdostal) [inline lib/publiccloud/gcp_client.pm:125]:
```suggestion
    return $self->gcr_zone . '/suse-sle-qa' . $self->project_id;
```

File: schedule/containers/sle_image_on_google_gke.yaml [hunk 3]
```diff
@@ -1,7 +1,8 @@
 ---
-name: sle_image_on_amazon_gke
+name: sle_image_on_google_gke
```
Comment (pdostal) [inline schedule/containers/sle_image_on_google_gke.yaml:2]: Please get rid of this file in favour of `lib/main_containers.pm` :smile:

File: tests/containers/push_container_image_to_gcr.pm [hunk 4]
```diff
@@ -40,7 +40,6 @@ sub post_fail_hook {
 
 sub post_run_hook {
     my ($self) = @_;
-    $self->{provider}->delete_image();
```
Comment (pdostal) [inline tests/containers/push_container_image_to_gcr.pm:43]: Why?

Comment (ilausuch) [reply tests/containers/push_container_image_to_gcr.pm:43]: We need to mantain this image on the repository to be used by the kubernetes test.

File: lib/publiccloud/basetest.pm (hunk 1 above)
Comment (ilausuch) [reply lib/publiccloud/basetest.pm:82]: We don't need to pass these parameters to have thihs working. See the creation of the GCE https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/f78569bc3e849bcbd1eb0478af11f8a87e357525/lib/publiccloud/basetest.pm#L86

File: lib/publiccloud/gke.pm [hunk 5]
```diff
@@ -0,0 +1,47 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Helper class for Google Container Registry (ECR)
```
Comment (jlausuch) [inline lib/publiccloud/gke.pm:6]:
```suggestion
# Summary: Helper class for Google Container Registry (GCR)
```

File: lib/publiccloud/gke.pm [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+has provider_client => undef;
+
+sub init {
+    my ($self, %args) = @_;
+    $self->SUPER::init("GKE");
+
+    my $cluster = get_var("PUBLIC_CLOUD_K8S_CLUSTER", "qe-c-testing");
+    assert_script_run("gcloud components install -q gke-gcloud-auth-plugin", 1200);
+    my $getCredentialsCommand = sprintf("gcloud container clusters get-credentials %s --zone %s", $cluster, "europe-west3");
```
Comment (jlausuch) [inline lib/publiccloud/gke.pm:24]: my $region = get_var('PUBLIC_CLOUD_REGION', 'europe-west3');
```suggestion
    my $getCredentialsCommand = sprintf("gcloud container clusters get-credentials %s --zone %s", $cluster, $region);
```

File: schedule/containers/sle_image_on_google_gke.yaml (hunk 3 above)
Comment (jlausuch) [reply schedule/containers/sle_image_on_google_gke.yaml:2]: Let's do that in another PR, as we have all the K8S schedules in `schedule/containers`, ok?

File: lib/publiccloud/gke.pm (hunk 6 above)
Comment (ilausuch) [reply lib/publiccloud/gke.pm]: done

File: schedule/containers/sle_image_on_google_gke.yaml (hunk 3 above)
Comment (pdostal) [reply schedule/containers/sle_image_on_google_gke.yaml:2]: Yes yes :smile:

**Review by pdostal — APPROVED** [review]

**Comment by ilausuch** [comment]
Last tests is using the new cloud kubernetes deployment

File: lib/publiccloud/gke.pm [hunk 7]
```diff
@@ -0,0 +1,48 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Helper class for Google Container Registry (GCR)
+#
+# Maintainer: Ivan Lausuch <ilausuch@suse.de>, qa-c team <qa-c@suse.de>
+# Documentation: https://docs.aws.amazon.com/AmazonECR/latest/userguide/docker-push-ecr-image.html
```
Comment (jlausuch) [inline lib/publiccloud/gke.pm:9]: wrong link

File: lib/publiccloud/gke.pm [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    my $cluster = get_var("PUBLIC_CLOUD_K8S_CLUSTER", "qe-c-testing");
+    my $zone = get_var("PUBLIC_CLOUD_ZONE", "europe-west4-a");
+    assert_script_run("gcloud components install -q gke-gcloud-auth-plugin", 1200);
+    my $getCredentialsCommand = sprintf("gcloud container clusters get-credentials %s --zone %s", $cluster, $zone);
+    assert_script_run($getCredentialsCommand, 120);
+}
+
+=head2 delete_container_image
+
+Clean a container image from the ECR
```
[the remark this answers is not in the document]
Comment (ilausuch) [reply lib/publiccloud/gke.pm]: Solved

File: lib/publiccloud/gke.pm (hunk 7 above)
Comment (ilausuch) [reply lib/publiccloud/gke.pm]: Solved

**Review by pdostal — APPROVED** [review]
Looks good and smooth

**Review by jlausuch — APPROVED** [review]
